Tests/MongoSwiftTests/ConnectionStringTests.swift

```diff
@@ -262,4 +262,39 @@ final class ConnectionStringTests: MongoSwiftTestCase {
             expect(try client.listDatabases().wait()).to(throwError())
         }
     }
+
+    func testLocalThresholdMSOption() throws {
+        // option is set correctly from options struct
+        let opts = MongoClientOptions(localThresholdMS: 100)
+        let connStr1 = try ConnectionString("mongodb://localhost:27017", options: opts)
+        expect(connStr1.options?["localthresholdms"]?.int32Value).to(equal(100))
+
+        // option is parsed correctly from string
+        let connStr2 = try ConnectionString("mongodb://localhost:27017/?localThresholdMS=100")
+        expect(connStr2.options?["localthresholdms"]?.int32Value).to(equal(100))
+
+        // options struct overrides string
+        let connStr3 = try ConnectionString("mongodb://localhost:27017/?localThresholdMS=50", options: opts)
+        expect(connStr3.options?["localthresholdms"]?.int32Value).to(equal(100))
+
+        let tooSmall = -10
+        expect(try ConnectionString(
```

Comment:
I'm actually not sure if we want to error here, the URI options spec says we should log a warning instead. though we don't have a way to do that now.
libmongoc also just seems to ignore but not log a warning in the case of a negative value here.

what do you think?

Reply:
Given our conversation, I think we decided on erroring, right?

Reply:
yup. since we also learned libmongoc does nothing on an invalid value of this option I've updated `ConnectionString` to inspect the value if its set via connection string and assert its non-negative.